### PR TITLE
Masonize

### PR DESCRIPTION
**Pull request by springmeyer**
Adapts node-osrm to the upcoming `masonize` work in osrm-backend: https://github.com/Project-OSRM/osrm-backend/pull/3119.

This simplifies the build. Now we depend on osrm-backend to:
- install mason (node-osrm needs to know nothing about mason)
- provide all dependencies and to control their versions

Before merging:
- [x] review
- [x] flip `osrm_release` value back to `master` from `masonize`.

/cc @TheMarex @daniel-j-h

### CODE REVIEW

**Comment by springmeyer** [comment]
This ready for review /cc @TheMarex @daniel-j-h

File: scripts/build.sh [hunk 1]
```diff
@@ -8,13 +8,30 @@ set -o pipefail
...
 export CURRENT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-source ${CURRENT_DIR}/travis_helper.sh
+export DEPS_DIR="$(pwd)/deps"
+export PATH=${DEPS_DIR}/bin:${PATH}
+mkdir -p ${DEPS_DIR}
+
+# install build deps from s3 (without mason client to keep things simple)
+# install cmake
+wget --quiet -O - https://mason-binaries.s3.amazonaws.com/${platform}-x86_64/cmake/${CMAKE_VERSION}.tar.gz | tar --strip-components=1 -xz -C ${DEPS_DIR} || exit 1
```
Comment (TheMarex) [inline scripts/build.sh]: Why is this simpler then `mason install`? Aren't we doing the same for osrm-backend now?
https://github.com/Project-OSRM/osrm-backend/blob/master/.travis.yml#L114

File: bootstrap.sh [hunk 2]
```diff
@@ -33,6 +26,7 @@ export TARGET_DIR=${TARGET_DIR:-$(pwd)/lib/binding}
 export OSRM_REPO=${OSRM_REPO:-"https://github.com/Project-OSRM/osrm-backend.git"}
 export OSRM_DIR=$(pwd)/deps/osrm-backend-${BUILD_TYPE}
 export JOBS=${JOBS:-1}
+export INSTALL_PREFIX=${INSTALL_PREFIX:-"/tmp/osrm-backend"}
```
Comment (TheMarex) [inline bootstrap.sh]: `INSTALL_PREFIX` could be confused with `TARGET_DIR`. `TMP_PREFIX` would maybe be better?

File: scripts/build.sh (hunk 1 above)
Comment (springmeyer) [inline scripts/build.sh]: Realizing the catch is that we need to set up the compiler in a way that both osrm-backend and node-osrm can get access to it. The current solution installs the compiler first, but before `osrm-backend` has been download, so we don't yet have access to the vendored mason command. Hrm...

File: bootstrap.sh (hunk 2 above)
Comment (springmeyer) [inline bootstrap.sh]: 👍  https://github.com/Project-OSRM/node-osrm/commit/b990a3a4ed36fab250854dc2e607b90e68da74cb

File: bootstrap.sh [hunk 3]
```diff
@@ -125,45 +87,20 @@ function build_osrm() {
...
-    ${MASON_HOME}/bin/cmake ../ -DCMAKE_INSTALL_PREFIX=${MASON_HOME} \
-      -DCMAKE_CXX_COMPILER="$CXX" \
-      -DBoost_NO_SYSTEM_PATHS=ON \
-      -DTBB_INSTALL_DIR=${MASON_HOME} \
-      -DCMAKE_CXX_FLAGS="${CXXFLAGS:-}" \
-      -DCMAKE_INCLUDE_PATH=${MASON_HOME}/include \
-      -DCMAKE_LIBRARY_PATH=${MASON_HOME}/lib \
+    cmake ../ -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} \
+      -DENABLE_MASON=ON \
+      -DCMAKE_CXX_COMPILER=${CXX} \
```
Comment (springmeyer) [inline bootstrap.sh]: https://github.com/Project-OSRM/node-osrm/commit/09c1ef2f1d1dff18441a0eaf6724272e84dbe144

**Pull request merged**